examples/vision/3D_image_classification.py

```diff
@@ -0,0 +1,454 @@
+a binary classification problem.
+"""
+
+url = "https://github.com/hasibzunair/3D-image-classification-tutorial/releases/download/v0.1/NSCLC-Radiomics-Genomics.zip"
+filename = os.path.join(os.getcwd(), "NSCLC-Radiomics-Genomics.zip")
+keras.utils.get_file(filename, url)
+
+with zipfile.ZipFile("NSCLC-Radiomics-Genomics.zip", "r") as z_fp:
+    z_fp.extractall("./")
+
+"""
+## Load data
+
+The files are provided in Nifti format with the extension .nii. To read the
+scans, the nibabel package is used. To process the data, the following
+is done:
+
+* Volumes are originally rotated by 90 degress, so the orientation is fixed
+* Resize width, height and depth
+
+Here several helper functions are defined to process the data. These functions
+will be used when building training and test datasets.
+"""
+
+import numpy as np
+import nibabel as nib
+import cv2
+
+from scipy.ndimage import zoom
+
+
+def read_nifti_file(filepath):
+    # read file
+    scan = nib.load(filepath)
+    # get raw data
+    scan = scan.get_fdata()
+    # rotate
+    scan = np.rot90(np.array(scan))
+    return scan
+
+
+def resize_slices(img):
+    # resize all slices
+    flatten = [
+        cv2.resize(img[:, :, i], (128, 128), interpolation=cv2.INTER_CUBIC)
+        for i in range(img.shape[-1])
+    ]
+    # stack along the z-axis
+    img = np.array(np.dstack(flatten))
+    return img
+
+
+def resize_depth(img):
+    # set the desired depth
+    desired_depth = 128
+    # get current depth
+    current_depth = img.shape[-1]
+    # compute depth factor
+    depth = current_depth / desired_depth
+    depth_factor = 1 / depth
+    # resize across z-axis
+    img_new = zoom(img, (1, 1, depth_factor), mode="nearest")
+    return img_new
+
+
+def process_scan(path):
+    # read scan
+    volume = read_nifti_file(path)
+    # resize width and height
+    volume = resize_slices(volume)
+    # resize across z-axis
+    volume = resize_depth(volume)
+    return volume
+
+
+"""
+Let's read the paths of the CT scans from the class directories
+"""
+
+# folder "1" consist of right CT scans
+right_scan_paths = [
+    os.path.join(os.getcwd(), "NSCLC-Radiomics-Genomics/1", x)
+    for x in os.listdir("NSCLC-Radiomics-Genomics/1")
+]
+# folder "2" consist of left CT scans
+left_scan_paths = [
+    os.path.join(os.getcwd(), "NSCLC-Radiomics-Genomics/2", x)
+    for x in os.listdir("NSCLC-Radiomics-Genomics/2")
+]
+
+print("CT scans with cancerous regions in left side: " + str(len(left_scan_paths)))
+print("CT scans with cancerous regions in right side: " + str(len(right_scan_paths)))
+
+"""
+Let's visualize a CT scan and also look at it's dimensions.
+"""
+
+import matplotlib.pyplot as plt
+
+img = read_nifti_file(right_scan_paths[2])
+print("Dimension of the CT scan is:", img.shape)
+plt.imshow(img[:, :, 5], cmap="gray")
+
+"""
+Since a CT scan has many slices, let's visualize a montage of the slices.
+"""
+
+
+def plot_slices(data):
+    # plot a montage of 20 CT slices
+    a, b = 2, 10
```

Comment:
Everywhere, please use fully-spelled out variable names for better readability

Reply:
Fixed!